src/main/java/frc/trigon/robot/commands/factories/AutonomousCommands.java

```diff
@@ -0,0 +1,107 @@
+    }
+
+    public static Command getFeedNoteCommand() {
+        return new ParallelCommandGroup(
+                IntakeCommands.getSetTargetStateCommand(IntakeConstants.IntakeState.FEED_SHOOTING),
+                GeneralCommands.getVisualizeNoteShootingCommand()
+        ).until(() -> !RobotContainer.INTAKE.hasNote());
+    }
+
+    public static Command getAlignToNoteCommand() {
+        return new InstantCommand(
+                () -> {
+                    NOTE_DETECTION_CAMERA.startTrackingBestObject();
+                    overrideRotation(AutonomousCommands::calculateRotationOverride);
+                }
+        ).andThen(getSetCurrentLEDColorCommand());
+    }
+
+    public static Command getStopAligningToNoteCommand() {
+        return new InstantCommand(() -> overrideRotation(Optional::empty));
+    }
+
+    public static Command getPrepareForShooterEjectionCommand(boolean isClose) {
+        return isClose ? getPrepareForCloseShooterEjectionCommand() : getPrepareForShooterEjectionCommand();
+    }
+
+    private static Command getPrepareForShooterEjectionCommand() {
+        return new ParallelCommandGroup(
+                PitcherCommands.getSetTargetPitchCommand(ShootingConstants.EJECT_FROM_SHOOTER_PITCH),
+                ShooterCommands.getSetTargetVelocityCommand(ShootingConstants.EJECT_FROM_SHOOTER_VELOCITY_ROTATIONS_PER_SECOND)
+        );
+    }
+
+    private static Command getPrepareForCloseShooterEjectionCommand() {
+        return new ParallelCommandGroup(
+                PitcherCommands.getSetTargetPitchCommand(ShootingConstants.CLOSE_EJECT_FROM_SHOOTER_PITCH),
+                ShooterCommands.getSetTargetVelocityCommand(ShootingConstants.CLOSE_EJECT_FROM_SHOOTER_VELOCITY_ROTATIONS_PER_SECOND)
+        );
+    }
+
+    private static Optional<Rotation2d> calculateRotationOverride() {
+        NOTE_DETECTION_CAMERA.trackObject();
+        if (RobotContainer.INTAKE.hasNote() || !NOTE_DETECTION_CAMERA.hasTargets())
+            return Optional.empty();
+
+        final Rotation2d currentRotation = RobotContainer.POSE_ESTIMATOR.getCurrentPose().getRotation();
+        final Rotation2d targetRotation = NOTE_DETECTION_CAMERA.getTrackedObjectYaw().minus(currentRotation);
+        return Optional.of(targetRotation);
+    }
+
+    private static void overrideRotation(Supplier<Optional<Rotation2d>> rotationOverride) {
+        PPHolonomicDriveController.setRotationTargetOverride(rotationOverride);
+    }
+
+    private static Command getSetCurrentLEDColorCommand() {
+        return GeneralCommands.getContinuousConditionalCommand(
+                LEDStripCommands.getStaticColorCommand(Color.kGreen, LEDStrip.LED_STRIPS),
+                LEDStripCommands.getStaticColorCommand(Color.kRed, LEDStrip.LED_STRIPS),
+                NOTE_DETECTION_CAMERA::hasTargets
+        ).asProxy().until(RobotContainer.INTAKE::hasNote);
```

Comment:
Aligning doesn't necessarily stop when we have a note, isn't that why you created the stop align command?